### DISCUSSION TITLE
Features requests

### DISCUSSION THREAD

**Discussion by allangood** [allangood/rtlamr2mqtt#64] [Ideas]
Please, use this thread to ask for new features.

**Comment by drmulligan** [comment]
Please add timestamps to each log line.  Timestamps make troubleshooting much easier.

**Reply by allangood** [reply]
Yes, this is something I should have implement a long time ago! :)
Let me put this to the next commit! :)

Thanks!

**Reply by drmulligan** [reply]
Excellent, thanks.

**Reply by allangood** [reply]
Added on #69 - Version 1.5.4

**Comment by drmulligan** [comment]
Please add device_class to the addon configuration.  Maybe it could be optional to not break existing installations?

**Reply by allangood** [reply]
I am already working on it! It is taking a little longer because I want to add it as a list of possible values.

Thank you!

**Reply by drmulligan** [reply]
Perfect, thank you.

**Reply by allangood** [reply]
Added on #69 - Version 1.5.4

**Comment by drmulligan** [comment]
rtlamr only runs every 5 minutes by default.  I would like to run rtl_443 in between at a different frequency than the  utility meter data, in this case at 433MHz to grab temperature and humidity data.  Any thoughts on how I should approach this?

**Reply by allangood** [reply]
I have seen people talking about using one antenna for multiple frequencies but they always have bad experiences. Using two antennas, one for each frequency is the best approach, otherwise packets would be missed.

But if you want to go forward with your idea, ideally you should run one RTL_TCP instance as a stand-alone app/container and then use  rtlamr2mqtt to connect to that remote instance. Check the "rtlamr_command" section for an example on how to do this.

**Comment by stu1811** [comment]
Can you add the ability to change the output units? Home Assistant expects water in m^3 but my meter output in ft^3. I currently have a template setup to do the conversion.

**Comment by allangood** [comment]
I'm not sure if I followed,  but this is what the unit_of_measurement was
made for.

**Reply by stu1811** [reply]
@allangood my meter outputs ft^3 but I want the the mqtt message into HA to be in m^3. I just found the other units discussion. I added my template there.  https://github.com/allangood/rtlamr2mqtt/discussions/68

**Reply by stu1811** [reply]
Did I misunderstand the unit of measure field? I thought it was to add units to the existing data and not convert it?

**Comment by allangood** [comment]
No you did not.

It really doesn't convert anything.

**Comment by drmulligan** [comment]
Please add package usbutils to the docker image.  Also please consider logging 'lsusb  |  grep -i rtl' on startup.

**Reply by allangood** [reply]
Hello @drmulligan , I am planning something different for the next release.
I really don't like to execute external (shell) commands and rather use native calls when possible.

This is what is planned for the next release:
https://github.com/pyusb/pyusb

The idea is to have an USB parameter to figure everything during start up.

**Comment by rafaqat75** [comment]
I have a Landis Gyr e470. Not on the compatible device list. Is there any chance this could be made to work with it?
https://www.landisgyr.eu/product/landisgyr-e470/

**Comment by melyux** [comment]
Please add a verbosity level that's even less verbose than info. In case no logging required for every reading, just on extraordinary events.

**Comment by dresslerc** [comment]
I have a Govee H5054 water leak sensor.  It works on another docker package: https://github.com/merbanan/rtl_433/issues/2189. But I am using your package for my water/gas meters.  Would love to have the Govee devices added to this package.

**Comment by gcormier** [comment]
Please allow selecting the RTLSDR device using serial number.

**Reply by HarrisonWilson** [reply]
Agree. If you plug in multiple dongles they will share the same ID, and the rtl_tcp command always adds -d 0 even if you manually supply that as a startup command in the config. The current mechanism to distinguish devices isn't helpful because it does not uniquely 2 rtl dongles since the vendor/device ID will be the same for two. 

At the moment, the only suitable workaround is to use your own standalone rtl_tcp.
